### PR TITLE
desktop: add appx target

### PR DESCRIPTION
**Pull request by sandymcfadden**
### Description

This updates the configuration to build an unsigned appx target of the app to submission to the Windows Store.
It also adds the needed logo assets.

### To test

Build on Windows from the desktop folder by running `make package`.
One of the files in the `release` folder will be an appx. You can double click but won't be able to install as it is unsigned. We'll need to send to the app store and download it through there to fully test the build and process.

### CODE REVIEW

File: desktop/electron-builder.json [hunk 1]
```diff
@@ -41,10 +41,25 @@
...
-		}
+		"icon": "./resource/icons/icon.ico",
+		"target": [
+			{
+				"target": "nsis",
+				"arch": [ "x64", "ia32" ]
+			},
+			{
+				"target": "appx",
+				"arch": [ "x64", "ia32" ]
```
Comment (nsakaimbo) [inline desktop/electron-builder.json]: Do we need both an x64 and ia32 build of the appx, or is a single ia32 installer sufficient? (Will work on both 32-bit and 64-bit systems). Do we submit both for Simplenote?

Comment (sandymcfadden) [reply desktop/electron-builder.json]: I'm not certain if we need to, we do submit both for Simplenote so I was following that here as well.

Comment (nsakaimbo) [reply desktop/electron-builder.json]: Makes sense - let's go with whatever we're doing for Simplenote currently.

File: desktop/electron-builder.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
 	"appId": "com.automattic.wordpress",
 	"asar": false,
 	"directories": {
-		"buildResources": "./resource/icons",
+		"buildResources": "./resource",
```
Comment (nsakaimbo) [inline desktop/electron-builder.json:6]: Why is this change necessary? Will it interfere with the way assets for the other Windows targets are located/generated?

Comment (sandymcfadden) [reply desktop/electron-builder.json:6]: We need this because the appx folder with resources for that target needs to be exposed in the root buildResources folder.

I updated the reference path to the icon file so it will be used properly still.

**Review by nsakaimbo — APPROVED** [review]
LGTM 👍

File: .circleci/config.yml [hunk 3]
```diff
@@ -547,6 +547,26 @@ jobs:
...
+      # - when:
+          # condition: << pipeline.git.tag >>
+          # steps:
+      - run:
+          name: Build Appx
+          command: |
+            # Note: Windows Store build should be unsigned
+            make -f desktop/Makefile package ELECTRON_BUILDER_ARGS='--config="electron-builder-appx.json"'
+      - run:
+          name: Clean Up Appx
```
Comment (nsakaimbo) [inline .circleci/config.yml:563]: @sandymcfadden This change seems to work to build the unsigned .appx. However there are a bunch of additional artifacts that are generated that I can't seem to delete. These cleanup commands work in a local Powershell instance and when I SSH into the Circle machine, however for some reason the additional artifacts aren't being cleaned up when the CI script actually runs.

File: .circleci/config.yml [hunk 4]
```diff
@@ -547,6 +547,28 @@ jobs:
...
+          steps:
+            - run:
+                name: Build Appx
+                command: |
+                  # Note: Windows Store build should be unsigned
+                  make -f desktop/Makefile package ELECTRON_BUILDER_ARGS='--config="electron-builder-appx.json"'
+            - run:
+                name: Clean Up Appx
+                command: |
+                  # FIXME: For some reason this the additional artifacts
```
Comment (nsakaimbo) [inline .circleci/config.yml:565]: Update: it seems the additional artifacts are being generated by the NSIS portion.

File: desktop/app/lib/config/index.js [hunk 5]
```diff
@@ -1,5 +1,5 @@
-const pkg = require( '../../../../desktop/package.json' );
-const config = require( '../../config.json' );
+const pkg = require( '../../../package.json' );
```
Comment (nsakaimbo) [inline desktop/app/lib/config/index.js:1]: @sandymcfadden I ended up catching a bug in the code - looks like these import paths were wrong (might have slipped through the cracks when we ported things over during the rewrite). 

It turns out we can use the generated config to override behaviors at run time. In fact, there already exists a check for whether an `updater` variable is set on the config. I override here it here in this file, based on the presence of a `WINDOWS_STORE` environment variable we can set at build time.

**Review by wp-desktop — DISMISSED** [review]
WordPress Desktop CI Failure for job "wp-desktop-mac".

@nsakaimbo please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/1042408). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

File: desktop/bin/build-desktop-config.js [hunk 6]
```diff
@@ -22,40 +22,11 @@ try {
...
-	process.exit( 1 );
+if ( process.env.WINDOWS_STORE ) {
+	// Disable auto and manual updates for the Windows Store
+	env = Object.assign( env, { updater: false } );
 }
 
-/*
- * Verify that the `config/secrets.json` file contents are valid JSON
- */
-let secretsObject;
```
Comment (nsakaimbo) [inline desktop/bin/build-desktop-config.js:44]: We're no longer authorizing via OAuth, so the remainder of this file is unnecessary.

File: desktop/bin/build-desktop-config.js [hunk 7]
```diff
@@ -22,40 +22,11 @@ try {
...
- * Verify that the `config/secrets.json` file exists. It contains OAuth keys that the app needs
- * to perform user login
- */
-let secretsString;
-try {
-	secretsString = fs.readFileSync( path.join( DESKTOP_DIR, 'config', 'secrets.json' ), 'utf-8' );
-} catch {
-	console.error( 'Config file config/secrets.json does not exist. Please create one.' );
-	process.exit( 1 );
+if ( process.env.WINDOWS_STORE ) {
```
Comment (nsakaimbo) [inline desktop/bin/build-desktop-config.js:27]: We can set a `WINDOWS_STORE` env var in Circle at build time, which will allow us override whether or not the updater is enabled.

**Comment by nsakaimbo** [comment]
_Update_: while testing this I noticed a critical bug that seemed to have slipped through the initial rewrite (glad we didn't deploy the new app yet!). TL;DR - the way the application config was generated was wrong, and I fixed the logic in the `build-desktop-config` script. We now make sure to call this script in each of the platform builds (Mac, Windows, Linux) prior to the application source being built. For Appx specifically, we set a `WINDOWS_STORE` environment variable so that the config is built with an `updater` value of `false`.

I downloaded a built Windows artifact and inspected the logs to ensure that the auto-updater is initialized for the NSIS build. We should probably take a quick gander at the Linux and Mac builds as well to verify. 

### To Test

Download the various platform builds and inspect the logs (Help -> Application Logs) to verify that:

- NSIS updater is initialized
- Appx updater is not initialized
- Mac updater is initialized
- Linux updater is initialized (note: linux does not have an auto-update mechanism per se - the best it can do is to tell you whether or not a newer version is available)

No need to generate the builds locally as all builds are available as artifacts of this PR since per the CI script `desktop/package.json` was changed (I amended one of the package scripts).
